src/frontend/hooks/server/projects.ts

```diff
@@ -9,3 +11,30 @@ export function useProject() {
   const projectContext = useActiveProjectContext();
   return projectContext.project;
 }
+
+export function useAllProjects() {
+  const api = useApi();
+
+  return useQuery({
+    queryFn: async () => await api.listProjects(),
+    queryKey: ['projects'],
+    throwOnError: true,
```

Comment:
mostly because i'm not super familiar, but can you provide some context for this? guessing it has something to do with relying on suspense or an error boundary instead of manually handling error states, but wondering what the criteria is for when you'd want this set. also, does it make sense for it to be the default for this hook, or an option to allow more flexibility case-by-case?

Reply:
Im going to get rid of this. I was playing around with Error Boundaries, but decided not to use this, and just forgot to take it out. As you said, this doesn't allow for flexibility with errors (it just makes all error one catch all), which is not the desired ui.

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/173/commits/1c9761ff81ed2cb6cfba0acfd6dddf374b5af0da